float64[] pos
float64[] pos2
float64[] ori
---
